### PR TITLE
Obsolete ToXmlString and FromXmlString on ECC types

### PR DESCRIPTION
**Pull request by vcsjones**
This obsoletes the `ToXmlString` and `FromXmlString` members as suggested at https://github.com/dotnet/runtime/issues/50528#issuecomment-897251402.

Contributes to #50528.

### CODE REVIEW

File: docs/project/list-of-diagnostics.md [hunk 1]
```diff
@@ -96,6 +96,7 @@ The PR that reveals the implementation of the `<IncludeInternalObsoleteAttribute
 |  __`SYSLIB0039`__ | TLS versions 1.0 and 1.1 have known vulnerabilities and are not recommended. Use a newer TLS version instead, or use SslProtocols.None to defer to OS defaults. |
 |  __`SYSLIB0040`__ | EncryptionPolicy.NoEncryption and AllowEncryption significantly reduce security and should not be used in production code. |
 |  __`SYSLIB0041`__ | The default hash algorithm and iteration counts in Rfc2898DeriveBytes constructors are outdated and insecure. Use a constructor that accepts the hash algorithm and the number of iterations. |
+|  __`SYSLIB0042`__ | ToXmlString and FromXmlString are obsolete on ECC types. Use a standard import and export format such as ExportSubjectPublicKeyInfo or ImportSubjectPublicKeyInfo for public keys and ExportPkcs8PrivateKey or ImportPkcs8PrivateKey for private keys. |
```
Comment (vcsjones) [inline docs/project/list-of-diagnostics.md:99]: @bartonjs I'm not thrilled with this wording, open to improvements. The main things I was trying to capture:

1. What they should be using instead

2. Doesn't affect To/FromXmlString on `RSA`.

Comment (bartonjs) [reply docs/project/list-of-diagnostics.md:99]: Maybe give some justification/motivation from the initial clause:

```suggestion
|  __`SYSLIB0042`__ | ToXmlString and FromXmlString have no implementation for ECC types, and are obsolete. Use a standard import and export format such as ExportSubjectPublicKeyInfo or ImportSubjectPublicKeyInfo for public keys and ExportPkcs8PrivateKey or ImportPkcs8PrivateKey for private keys. |
```

?

Comment (vcsjones) [reply docs/project/list-of-diagnostics.md]: I think that is what I felt was "missing". Thanks!

**Review by bartonjs — APPROVED** [review]

**Comment by bartonjs** [comment]
Opened paperwork at https://github.com/dotnet/docs/issues/28834

**Pull request merged**